### ISSUE TITLE
Suppressing output

### ISSUE THREAD

**Issue by prydonius** [prydonius/spinning_cursor#7]
Although print statements in actions don't cause any _fatal_ issues, they disrupt the cursor 'animation'.

e.g.

``` ruby
SpinningCursor.start
puts "Hello"
sleep 2
puts "Disrupting"
sleep 2
puts "the cursor :("
SpinningCursor.stop
```

Results in the following output:

```
Hello
Loading \\Disrupting
Loading \\the cursor :(
Done
```

We need to suppress it and either:
1. Spit all the buffered outputs at the end
2. Save output to a file (e.g. adding an option 'file')
3. Just hide completely (e.g. adding an option 'silent')

All three options should be available (at least 1 and 3) and it should default to 1.

Capturing output can be done similar to the method used in the tests.

**Comment by abinoam** [comment]
This issue is not that simple.
I took a look at it.
The problem is 2 threads writing to the same buffer ($stdout) without any kind of signaling/semaphore/mutex/? control between them.

I would take one of 2 ways:
1) Make the writing "thread-safe". The main thread (where action is being run) writes to a StringIO as you suggested. And the "spin" thread just "check" if there's something written there and arrange to write it the right way.
2) Use something like curses to better cursor positioning and handling these writes.

status: postponed because not that fast (for me)

**Comment by prydonius** [comment]
Fixed in PR #13, build 4902633.
